api/types/statichostuser/statichostuser.go

```diff
@@ -0,0 +1,86 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package statichostuser
+
+import (
+	"strconv"
+
+	headerv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/header/v1"
+	statichostuserpb "github.com/gravitational/teleport/api/gen/proto/go/teleport/statichostuser/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/trace"
+)
+
+// NewStaticHostUser creates a new host user to be applied to matching SSH nodes.
+func NewStaticHostUser(name string, spec *statichostuserpb.StaticHostUserSpec) (*statichostuserpb.StaticHostUser, error) {
+	hostUser := &statichostuserpb.StaticHostUser{
+		Kind:    types.KindStaticHostUser,
+		Version: types.V1,
+		Metadata: &headerv1.Metadata{
+			Name: name,
+		},
+		Spec: spec,
+	}
+	if err := ValidateStaticHostUser(hostUser); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return hostUser, nil
+}
+
+func isValidUidOrGid(s string) bool {
+	// No uid/gid is OK
+	if s == "" {
+		return true
+	}
+	// If uid/gid is present, it must be an integer (uid/gid are strings instead
+	// of ints to match user traits).
+	_, err := strconv.Atoi(s)
+	return err == nil
+}
+
+// ValidateStaticHostUser checks that required parameters are set for the
+// specified StaticHostUser.
+func ValidateStaticHostUser(u *statichostuserpb.StaticHostUser) error {
+	if u == nil {
+		return trace.BadParameter("StaticHostUser is nil")
+	}
+	if u.Metadata == nil {
+		return trace.BadParameter("Metadata is nil")
+	}
+	if u.Spec == nil {
+		return trace.BadParameter("Spec is nil")
+	}
+	if u.Spec.Login == "" {
+		return trace.BadParameter("missing login")
+	}
+	if u.Spec.NodeLabels != nil {
+		for key, value := range u.Spec.NodeLabels.Values {
+			if key == types.Wildcard && !(len(value.Values) == 1 && value.Values[0] == types.Wildcard) {
+				return trace.BadParameter("selector *:<val> is not supported")
+			}
+		}
+	}
```

Comment:
Trait templating doesn't apply here because when we create a host user, there won't be a user to pull traits from. I tried adding the expression validation, but it would require moving a LOT of stuff from `lib/utils` to `api/utils` and I think it would be much simpler to just do that in `lib/services` like we do for roles.